### ISSUE TITLE
TFLite produces bogus results on GPU when setPrecisionLossAllowed(true)

### ISSUE THREAD

**Issue by bmmo** [tensorflow/tensorflow#48671]
### 1. System information

- OS Platform and Distribution: Android 10
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.4.0

### 2. Code

I am using a FC-DenseNet model trained with PyTorch, then converted to ONNX, Tensorflow .pb and finally TFLite.
Original torch code is here: https://github.com/petko-nikolov/pysemseg/blob/master/pysemseg/models/densenet.py

Additionally, I added a `permute` operation (equivalent to Numpy transpose), `Argmax` and conversion to `np.uint8` in the end of the model. 

### 3. Failure after conversion
Model converts **OK**.
Python tests on Desktop are **OK** both with Tensorflow and TFLite.
Model is **OK** on Android using CPU
Model is **NOT OK** on Android using GPU Delegate
Model is **OK** on Android using GPU Delegate and `setPrecisionLossAllowed(false)`

**Comment by abattery** [comment]
@impjdi could you take a look?

**Comment by impjdi** [comment]
The title of the bug and 

> Model is OK on Android using GPU Delegate and `setPrecisionLossAllowed(false)`

are telling contradicting story.  Which one is it?

Also, could you attach the model?

**Comment by bmmo** [comment]
@impjdi the title was wrong and was fixed.
I'm not sure I can attach the model. Can I give you any other relevant info without providing the model?

**Comment by bmmo** [comment]
[quantized-fp16.tflite.zip](https://github.com/tensorflow/tensorflow/files/6358625/quantized-fp16.tflite.zip)

I'm attaching the same model, quantised to FP16, but with different weights. Hope this helps? 
Anyway let me know what more information you would need.

**Comment by misheeltoli** [comment]
> [quantized-fp16.tflite.zip](https://github.com/tensorflow/tensorflow/files/6358625/quantized-fp16.tflite.zip)
> 
> I'm attaching the same model, quantised to FP16, but with different weights. Hope this helps? Anyway let me know what more information you would need.

I have a same problem with setPrecisionLossAllowed option. @bmmo Have you found the solution?

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.

**Comment by impjdi** [comment]
Sorry, this dropped off my radar.

There are networks which require higher precision and therefore can't use FP16.  See the range FP16 can represent: https://en.wikipedia.org/wiki/Half-precision_floating-point_format  You have to check whether your model doesn't run into range issues with FP16.  Sometimes, it's possible to work around this issue by normalization between layers and using bounded activation functions such as ReLU6 instead of ReLU, but these are not a universal answer and can be case by case.
